verify.go

```diff
@@ -87,6 +87,8 @@ type Config struct {
 	// If true, token expiry is not checked.
 	SkipExpiryCheck bool
 
+	IssuerChecker func(issuer string) bool
```

Comment:
Instead of this, let's do `SkipIssuerCheck` and add a good doc string:

```
// SkipIssuerCheck is intended for specialized cases where the the caller wishes to defer
// issuer validation. When enabled, callers MUST independently verify the Token's Issuer
// is a known good value.
//
// Mismatched issuers often indicate client mis-configuration. If mismatches are unexpected,
// evaluate if the provided issuer URL is incorrect instead of enabling this option.
SkipIssuerCheck bool
```